library/src/hcc_detail/hipsolver.cpp

```diff
@@ -12,7 +12,9 @@
 
 using namespace std;
 
+#ifdef __cplusplus
 extern "C" {
+#endif
```

Comment:
You don't need that check here. That construct is used in files that are compiled as either C or as C++. Generally speaking, that check is just used `*.h` files which need to support getting included into both `*.c` and `*.cpp` files.